### ISSUE TITLE
Unable to automate use of the "More" menu of Horizontal NavigationView

### ISSUE THREAD

**Issue by mrlacey** [microsoft/microsoft-ui-xaml#2736]
**Describe the bug**

When using a `NavigationView` in horizontal mode and with enough items included that the "More" menu option is displayed, it is not possible to use automation tools to interact with the contents of the menu displayed when clicking the "More" button.

Is automated interaction with the menu supported?
If not supported, is there a reason?
If it is supported, how has it been tested?

**Context** 
I'm using automated tests (powered by WinAppDriver) to walk the entire navigation structure of an app to test that everything is as expected and there are no obvious errors. As the apps in question now include enough options to necessitate the "More" option/menu, it's no longer possible to test the apps fully.

I can query how many items are in the pop-up, but querying details (such as the text) or trying to click on an item fails.
The pop-up seems to be being created in the same way as the menu items in a MenuBar and I can query and interact with menubar items without issue. I'm assuming there is a non-obvious difference in the way the pop-up is created or there's something special about the way that NavigationViewItems are displayed in the pop-up.

**Steps to reproduce the bug**

1. Add a NavigationView with `PaneDisplayMode="Top"`
2. Add enough items that the "More" menu will be displayed.
3. Use automation tools to open the menu and query the contents of the list. Example code below.

```cs
var moreBtn = appSession.FindElementByName("More");
moreBtn.Click();
var popupMenu = appSession.FindElementByName("Pop-up");

var moreMenuItemsCount = popupMenu.FindElementsByClassName("Microsoft.UI.Xaml.Controls.NavigationViewItem").Count;
moreBtn.Click(); // Close menu so get in a consistent state

for (int i = 0; i < moreMenuItemsCount; i++)
{
    moreBtn.Click();

    // Find the pop-up again as it's recreated after opening the menu each time
    popupMenu = appSession.FindElementByName("Pop-up");

    var moreMenuItems = popupMenu.FindElementsByClassName("Microsoft.UI.Xaml.Controls.NavigationViewItem");

    // BOTH/EITHER OF THE FOLLOWING FAIL!
    var entryText = moreMenuItems[i].Text;
    moreMenuItems[i].Click();

    ....  // other code omitted
}
```

**Expected behavior**

It should be possible to query the contents of the pop-up menu in the same way as any other content. (This could be any content on the main window or in other pop-ups line menu flyouts.)

**Screenshots**

This menu:

![image](https://user-images.githubusercontent.com/189547/85416129-04338b80-b566-11ea-909e-7483dfa36f1d.png)

**Version Info**

NuGet package version: 

[Microsoft.UI.Xaml 2.4.2]

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (19621)              |  Yes   |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            |  Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

Related issue also raised on the WinAppDriver repository: https://github.com/microsoft/WinAppDriver/issues/1204

**Comment by StephenLPeters** [comment]
I think that this is effecting Narrator navigation within this popup as well

**Comment by StephenLPeters** [comment]
@ojhad

**Comment by ojhad** [comment]
> Is automated interaction with the menu supported?

Yeah this is definitely an issue and should be supported. 

Our current tests that interact with overflow items look up the elements via the Name property and invoke using `.Click()`.

**Comment by mrlacey** [comment]
> Our current tests that interact with overflow items look up the elements via the Name property and invoke using `.Click()`.

That's great for fixed items, but doesn't allow for dynamic content.

**Comment by ojhad** [comment]
> That's great for fixed items, but doesn't allow for dynamic content.

Yeah for sure. As part of the PR for this fix, I think we should add a test to reflect this scenario.

Just as an FYI, I did a very quick scan of our current test corpus and found one instance where we retrieve a list of items from the overflow menu and invoke an item there. [We don't seem to be using `.click` in this scenario](https://github.com/microsoft/microsoft-ui-xaml/blob/5fcc4ed085983526944b70dedd4d59d1d60ac747/dev/NavigationView/NavigationView_InteractionTests/NavigationViewTestsBase.cs#L313)

**Comment by mrlacey** [comment]
> Just as an FYI, I did a very quick scan of our current test corpus and found one instance where we retrieve a list of items from the overflow menu and invoke an item there. [We don't seem to be using `.click` in this scenario](https://github.com/microsoft/microsoft-ui-xaml/blob/5fcc4ed085983526944b70dedd4d59d1d60ac747/dev/NavigationView/NavigationView_InteractionTests/NavigationViewTestsBase.cs#L313)

Wow. Having spent some time looking over the test code, it works by making a copy of all the items in the Windows UI Hierarchy, querying the copy collection/cache and then creating a new item based on the cache which is invoked.
This works when the tests and tested code are in the same application but is not sufficient to verify that the control can be used through automation tooling.

**Comment by mrlacey** [comment]
accessibility issues shouldn't be closed without good reason
